src/main/java/seedu/address/model/date/Date.java

```diff
@@ -25,27 +25,26 @@ public class DateOfBirth implements Comparable<DateOfBirth> {
     //for changing to user-readable format
     private static final DateTimeFormatter outputFormatter = DateTimeFormatter.ofPattern("MMM d, yyyy");
 
-
     public final LocalDate date;
 
     /**
-     * Constructs an {@code DateOfBirth}.
+     * Constructs a {@code Date}.
      *
      * @param date A valid date.
      */
-    public DateOfBirth(String date) {
+    public Date(String date) {
         requireNonNull(date);
-        checkArgument(isValidDateOfBirth(date), MESSAGE_CONSTRAINTS);
+        assert !date.isBlank();
+        checkArgument(isValidDate(date), MESSAGE_CONSTRAINTS);
         this.date = LocalDate.parse(date, logFormatter);
     }
 
     /**
-     * Returns true if a given string is a valid DOB input.
+     * Returns true if a given string is a valid Date input.
      * @return boolean
      */
-
     //found from https://mkyong.com/java/how-to-check-if-date-is-valid-in-java/
```

Comment:
Just a minor modification to follow the code reuse guidelines
```suggestion
    //Solution below adapted from https://mkyong.com/java/how-to-check-if-date-is-valid-in-java/
```